### PR TITLE
WiP: ACS-2493 LocalTransform request using a DAU

### PR DESCRIPTION
**Pull request by kmagdziarz**

### CODE REVIEW

File: repository/src/main/java/org/alfresco/repo/rendition2/LocalSynchronousTransformClient.java [hunk 1]
```diff
@@ -51,12 +57,24 @@
     private static Log logger = LogFactory.getLog(LocalTransformClient.class);
 
     private LocalTransformServiceRegistry localTransformServiceRegistry;
+    private ContentService contentService;
+    private Boolean isDirectAccessUrlEnabled;
```
Comment (alandavis) [inline repository/src/main/java/org/alfresco/repo/rendition2/LocalSynchronousTransformClient.java:61]: could you make this a lowercase boolean? - code change in setter method

File: repository/src/main/java/org/alfresco/repo/rendition2/LocalSynchronousTransformClient.java [hunk 2]
```diff
@@ -145,4 +165,18 @@ public String getName()
     {
         return "Local";
     }
+
+    private void setDirectAccessUrlIfEnabled(Map<String, String> actualOptions,
+                                             NodeRef sourceNodeRef,
+                                             String localTransformName)
+    {
+        if (isDirectAccessUrlEnabled
```
Comment (alandavis) [inline repository/src/main/java/org/alfresco/repo/rendition2/LocalSynchronousTransformClient.java:173]: In the repo && and || tend to be at the end of the line

File: repository/src/main/resources/alfresco/repository.properties [hunk 3]
```diff
@@ -1012,6 +1012,10 @@ local.transform.service.enabled=true
 local.transform.service.cronExpression=4 30 0/1 * * ?
 local.transform.service.initialAndOnError.cronExpression=0 * * * * ?
```
Comment (alandavis) [inline repository/src/main/resources/alfresco/repository.properties:1014]: Change comment: A switch to enable or disable the use of Direct Access URLs in transforms. Must also be supported for the source node.

File: repository/src/main/java/org/alfresco/repo/rendition2/LocalTransformClient.java [hunk 4]
```diff
@@ -145,7 +155,7 @@ public void transform(NodeRef sourceNodeRef, RenditionDefinition2 renditionDefin
                             throw new IllegalStateException("isSupported was not called prior to an asynchronous transform.");
                         }
```
Comment (alandavis) [inline repository/src/main/java/org/alfresco/repo/rendition2/LocalTransformClient.java:157]: When using DUAs, we will not be using the contentReader. You may wish to null it out or just ignore it later on. If you do just ignore it, you will need to remove the DIRECT_ACCESS_URL from the options once the first step in a pipeline has been executed.

Will need to do something similar in the LocalSynchronousTransformClient class, but there avoiding creating a reader is harder.

Comment (kmagdziarz) [reply repository/src/main/java/org/alfresco/repo/rendition2/LocalTransformClient.java:157]: `                        
setDirectAccessUrlIfEnabled(actualOptions, sourceNodeRef, localTransform);

localTransform.transform(reader, writer, actualOptions, renditionName, sourceNodeRef);

removeDirectAccessUrlAfterTransform(actualOptions);
` 
line 171, removing DAU after transform request, is it sufficient?

**Comment by alandavis** [comment]
Closing this PR - See #988

**Pull request closed without merging**